src/libaktualizr/uptane/uptane_init_test.cc

```diff
@@ -261,6 +261,44 @@ TEST(Uptane, InitializeFail) {
   }
 }
 
+/**
+ * Verifies if the system hostname is used as a primary ECU hardware ID
+ * if it's not specified in the configuration
+ *
+ * Checks actions:
+ *
+ * - [x] Use the system hostname as hardware ID if one is not provided
+ */
+TEST(Uptane, HostnameAsHardwareID) {
+  TemporaryDirectory temp_dir;
+  Config conf("tests/config/basic.toml");
+  conf.storage.path = temp_dir.Path();
+
+  boost::filesystem::copy_file("tests/test_data/cred.zip", temp_dir.Path() / "cred.zip");
+  conf.provision.provision_path = temp_dir.Path() / "cred.zip";
+
+  {
```

Comment:
We should explicitly check after constructing the config but before initialization that the config's `provision.primary_ecu_hardware_id` field is empty.

Reply:
Agree